lua/dashboard/session.lua

```diff
@@ -35,33 +30,81 @@ function session.session_save(name)
   end
 
   local file_name = name == nil and project_name() or name
-  local file_path = db.session_directory .. '/' .. file_name .. '.vim'
+  local file_path = db.session_directory .. path_sep .. file_name .. '.vim'
   api.nvim_command('mksession! ' .. fn.fnameescape(file_path))
   vim.v.this_session = file_path
-  vim.notify('Session ' .. file_name .. ' is now persistent')
+
+  if db.session_verbose then
+    vim.notify('Session ' .. file_name .. ' is now persistent')
+  else
+    vim.notify('This session is now persistent')
+  end
+  session_cp_name = project_name()
 end
 
 function session.session_load(name)
   local file_name = name == nil and project_name() or name
-  local file_path = db.session_directory .. '/' .. file_name .. '.vim'
+  local file_path = db.session_directory .. path_sep .. file_name .. '.vim'
 
   if vim.v.this_session ~= '' and fn.exists('g:SessionLoad') == 0 then
     api.nvim_command('mksession! ' .. fn.fnameescape(vim.v.this_session))
   end
 
-  if fn.filereadable(file_path) then
+  if fn.filereadable(file_path) == 1 then
     vim.cmd([[ noautocmd silent! %bwipeout!]])
     api.nvim_command('silent! source ' .. file_path)
 
     if vim.opt.laststatus:get() == 0 then
       vim.opt.laststatus = 2
     end
 
-    vim.notify('Loaded ' .. file_path .. ' session')
+    if db.session_verbose then
```

Comment:
as i said this check it's duplicate . i think there should be have a better design to avoid some duplicate codes.

Reply:
@glepnir I changed this to use a message dictionary.